auctioneer/client.go

```diff
@@ -464,14 +464,18 @@ func (c *Client) SubmitOrder(ctx context.Context, o order.Order,
 	// Split into server message which is type specific.
 	switch castOrder := o.(type) {
 	case *order.Ask:
-		constraints := auctioneerrpc.ChannelAnnouncementConstraints(
+		announcement := auctioneerrpc.ChannelAnnouncementConstraints(
 			castOrder.AnnouncementConstraints,
 		)
+		confirmations := auctioneerrpc.ChannelConfirmationConstraints(
```

Comment:
I think we should add an extra check in `SubmitOrder` that the channel type and confirmation constraints are compatible. For example I cannot use an explicit legacy channel with explicit zero conf.

Reply:
And in addition, I think it would make sense to also make sure that `determineBatchVersion() >= ZeroConf` if either the `bid.ZeroConf` or the `ask.ConfirmationConstraints = ZeroConfOnly`.

Reply:
Good call, checks for unannounced channels added too :+1: